ValidateJobs.py

```diff
@@ -0,0 +1,143 @@
+        print(f'\tDuplicate company url found on line {event.start_mark.line}:    {event.value}\n')
+    if type(company) == yaml.events.MappingEndEvent:
+      event = next(companiesGenerator)
+      if type(event) == yaml.events.ScalarEvent and event.value not in keys:
+        keys.append(event.value)
+      elif type(event) == yaml.events.ScalarEvent:
+        errorCount += 1
+        if not firstDuplicateFound:
+          firstDuplicateFound = True
+          print('There are duplicates found in the companies.yml file:')
+        print(f'\tDuplicate company key found on line {event.start_mark.line}:    {event.value}\n')
+  return errorCount
+
+
+def validateAgainstCompaniesList(companies, jobs, jobsGenerator, addNewline):
+  missing = [j['company'] for j in jobs if j['company'] not in companies]
+  
+  if len(missing) > 0:
+    if addNewline:
+      print('\n')
+    print('Cannot find the following companies in the companies.yml file:')
+    
+    event = next(jobsGenerator)
+    while(type(event) != yaml.events.StreamEndEvent):
+      if type(event) == yaml.events.ScalarEvent and event.value in missing:
+        print(f'\tLine {event.start_mark.line+1} of jobs.yml:    {event.value}\n')
+      event = next(jobsGenerator)
+  return len(missing)
+
+
+def checkForDuplicateCompanies(jobsGenerator, addNewline):
+  companies = []
+  errorCount = 0
+  firstDuplicateFound = False
+  for job in jobsGenerator:
+    if type(job) == yaml.events.ScalarEvent and job.value == 'company':
+      nextValue = next(jobsGenerator)
+      if nextValue.value not in companies:
+        companies.append(nextValue.value)
+      else:
+        errorCount += 1
+        if not firstDuplicateFound:
+          if addNewline:
+            print('\n')
+          firstDuplicateFound = True
+          print('There are duplicate companies in the jobs.yml file:')
+        print(f'\tDuplicate company found on line {nextValue.start_mark.line+1}:    {nextValue.value}\n')
+  return errorCount
+
+
+def checkForDuplicateLinks(jobsGenerator, addNewline):
+  urls = []
+  firstDuplicateFound = False
+  company = ''
+  errorCount = 0
+  for job in jobsGenerator:
+    if type(job) == yaml.events.ScalarEvent and job.value == 'company':
+      company = next(jobsGenerator).value
+    if type(job) == yaml.events.ScalarEvent and job.value == 'link':
+      nextValue = next(jobsGenerator)
+      if nextValue.value not in urls:
+        urls.append(nextValue.value)
+      else:
+        errorCount += 1
+        if not firstDuplicateFound:
+          if addNewline:
+            print('\n')
+          firstDuplicateFound = True
+          print('There are duplicate urls in the jobs.yml file:')
+        print(f'\tDuplicate link found for:    {company}\n\t\tLine {nextValue.start_mark.line+1}: {nextValue.value}\n')
+  return errorCount
+
+
+if __name__ == '__main__':
+
+  jobsFile = '_data/jobs.yml'
+  companiesFile = '_data/companies.yml'
```

Comment:
Can these be relative to a path passed as the first argument?

Would be nice to be able to run it like this:
```
python ValidateJobs.py /path/to/_data
```

Reply:
So you want to just pass a relative path? Would you wanna pass an absolute path? Since it's just the path I will still look for those specific files, right? Would you want it to default to the path used if nothing is provided or fail?

Reply:
Yeah, default to the current working directory.